### PR TITLE
Archive build logs daily

### PR DESCRIPTION
**Pull request by davemcorwin**
- task for archiving logs on a daily basis
- task for migrating historical logs
- update internal docs
- upgrade eslint-plugin-react for some reason

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -295,23 +285,46 @@ workflows:
...
                 - staging
                 - main
+
     jobs:
-      - nightly-builds:
+      - run-task:
+          name: archive-build-logs-staging
+          app: federalistapp-staging
+          task-name: archive-build-logs-$(date "%Y-%m-%d")
+          task-command: yarn archive-build-logs-daily $(date "%Y-%m-%d")
```
Comment (davemcorwin) [inline .circleci/config.yml:300]: not sure if this date thing is gonig to work

File: scripts/archive-build-logs-daily.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          [Op.gte]: startDate.toDate(),
+          [Op.lt]: endDate.toDate(),
+        },
+      },
+    });
+
+    console.log(`\nFound ${builds.length} builds.`);
+
+    const { errors } = await PromisePool
+      .withConcurrency(5)
```
Comment (davemcorwin) [inline scripts/archive-build-logs-daily.js:33]: Maybe we can handle more concurrent requests?

File: .circleci/config.yml (hunk 1 above)
Comment (apburnes) [reply .circleci/config.yml:300]: You could just retrieve the date in the function during execution since its only run once a day for that day.

**Review by apburnes — APPROVED** [review]
🚀

File: scripts/archive-build-logs-daily.js (hunk 2 above)
Comment (apburnes) [reply scripts/archive-build-logs-daily.js:33]: Once it's up and running, we should have a better idea on tuning concurrency.

**Comment by davemcorwin** [comment]
@apburnes yeah, we could add a default date if none provided , but i'm curious to see if this works.

**Pull request merged**